lib/virtus/support/type_lookup.rb

```diff
@@ -84,7 +82,7 @@ def determine_type_from_primitive(primitive)
       descendants.select(&:primitive).reverse_each do |descendant|
         descendant_primitive = descendant.primitive
         next unless primitive <= descendant_primitive
-        type = descendant if type.nil? or type.primitive > descendant_primitive
+        type = descendant if type.nil? || type.primitive > descendant_primitive
```

Comment:
Yes, I know that, but that is irrelevant in this case so I didn't feel the need to even add that bit.